### PR TITLE
CCU-194: Per-client mutes for Dolby Voice Server

### PR DESCRIPTION
**Pull request by matthiasha**

### CODE REVIEW

File: src/app/actions/ConferenceActions.js [hunk 1]
```diff
@@ -765,9 +765,10 @@ export class Actions {
...
       } = getState();
-      let user = VoxeetSDK.session.participant;
+      let user;
       if (!userId) {
-        userId = VoxeetSDK.session.participant.id;
+        user = VoxeetSDK.session.participant;
+        userId = user.id;
         isMuted = controls.isMuted;
       } else {
         user = VoxeetSDK.conference.participants.get(userId);
```
Comment (matthiasha) [inline src/app/actions/ConferenceActions.js:774]: trying to make this more readable

File: src/app/actions/ConferenceActions.js [hunk 2]
```diff
@@ -807,7 +808,17 @@ export class Actions {
...
       } else {
-        VoxeetSDK.conference.mute(user, isMuted ? false : true);
+        if (userId === VoxeetSDK.session.participant.id) {
+          VoxeetSDK.conference.mute(user, !isMuted);
+        } else {
+          let promise;
+          if (isMuted) {
+            promise = VoxeetSDK.conference.startAudio(user);
+          } else {
+            promise = VoxeetSDK.conference.stopAudio(user);
```
Comment (matthiasha) [inline src/app/actions/ConferenceActions.js:818]: always using `start/stopAudio()`, also for non-DVCS conferences, as discussed with Gary

File: src/app/actions/ConferenceActions.js [hunk 3]
```diff
@@ -807,7 +808,17 @@ export class Actions {
...
+          }
+          return promise.then(() => dispatch(ParticipantActions.onToogleMicrophone(userId)));
```
Comment (matthiasha) [inline src/app/actions/ConferenceActions.js:820]: I hope this makes sense. Intention is, wait for a successful change before triggering the event.

File: src/app/components/attendees/AttendeesList.js [hunk 4]
```diff
@@ -311,7 +311,7 @@ class AttendeesList extends Component {
                             <span className="participant-username">
                               {participant.name}
                             </span>
-                            {toggleMicrophone != null && !participant.isMyself && !dolbyVoiceEnabled && (
+                            {toggleMicrophone != null && !participant.isMyself && !(dolbyVoiceEnabled && currentUser.isListener) && (
```
Comment (matthiasha) [inline src/app/components/attendees/AttendeesList.js:314]: listeners on DVCS conferences will not be able to configure per-client mutes

File: src/app/components/attendees/modes/List.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ class List extends Component {
       kickParticipant,
       isAdmin,
       isAdminActived,
+      currentUser  // TODO no usages found for this component
```
Comment (matthiasha) [inline src/app/components/attendees/modes/List.js:18]: @Energizz please help - I added `currentUser` here but found no usages for the `List` component. If it is used somewhere, we need to pass in `currentUser`.

File: src/app/components/attendees/AttendeesSettings.js [hunk 6]
```diff
@@ -210,7 +210,7 @@ class AttendeesSettings extends Component {
     VoxeetSDK.mediaDevice.selectAudioInput(deviceId).then(() => {
       Cookies.set("input", deviceId, default_cookies_param);
       if (this.props.microphoneMuted) {
-        VoxeetSDK.conference.toggleMute(VoxeetSDK.session.participant);
+        VoxeetSDK.conference.mute(VoxeetSDK.session.participant, true);
```
Comment (matthiasha) [inline src/app/components/attendees/AttendeesSettings.js:213]: `toggleMute()` is undocumented. Replaced with `mute()`.

File: src/app/actions/ConferenceActions.js (hunk 1 above)
Comment (Energizz) [reply src/app/actions/ConferenceActions.js:774]: Fine for me

**Comment by Energizz** [comment]
@matthiasha Be careful, your MR is for master ⚠️ the feature should be merged into develop first

**Comment by matthiasha** [comment]
This PR was based on the wrong branch, please ignore. It was replaced by #111

**Pull request closed without merging**